test/test_optim.py

```diff
@@ -779,6 +780,51 @@ def _test_derived_optimizers(self, optimizer_pairs_with_flags, flag):
                 for k in st_p_state:
                     self.assertEqual(st_p_state[k], mt_p_state[k])
 
+    # ref: https://github.com/pytorch/pytorch/issues/100701
+    def _test_derived_optimizers_with_zero_size_params(self, optimizer_pairs_with_flags, flag):
```

Comment:
This should be merged with _test_derived_optimizers by adding a zero tensor to the params of the model there, since we know the foreach=False works as a good reference.